File: UI/grooveon_mobile/lib/models/player_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'player_response.g.dart';

@JsonSerializable()
class PlayerResponse {
  final int id;
  final int songId;
  final String title;
  final String artistName;
  final int duration;
  final String coverUrl;
  final String externalTrackId;
  final bool hasPrevious;
  final bool hasNext;

  PlayerResponse({
    required this.id,
    required this.songId,
    required this.title,
    required this.artistName,
    required this.duration,
    required this.coverUrl,
    required this.externalTrackId,
    required this.hasPrevious,
    required this.hasNext,
  });

  factory PlayerResponse.fromJson(Map<String, dynamic> json) =>
      _$PlayerResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PlayerResponseToJson(this);
}

File: GrooveOn.Models/ResponseObjects/PlayerResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
   public class PlayerResponse
{
    public int Id { get; set; }
    public int SongId { get; set; }
    public string Title { get; set; }
    public String ArtistName {get; set;}
    public int Duration {get; set;}
    public string CoverUrl { get; set; }
    public string? PreviewUrl { get; set; }
    public string ExternalTrackId { get; set; }
    public bool HasPrevious { get; set; }
    public bool HasNext { get; set; }
}
}
